modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/AbstractKeyValueStorage.java

```diff
@@ -104,6 +124,13 @@ protected AbstractKeyValueStorage(String nodeName, FailureManager failureManager
     /** Returns key values by revision for operation. */
     protected abstract Value valueForOperation(byte[] key, long revision);
 
+    /**
+     * Returns {@code true} if the storage is in recovery state and the watches have not {@link #startWatches started}.
+     *
+     * <p>Method is expected to be invoked under {@link #rwLock}.</p>
+     */
+    protected abstract boolean isRecoveryState();
```

Comment:
I think it would be better to call this method `inRecoveryState` or `isInRecoveryState`

Reply:
Thanks.